diff --git a/node_modules/react-notion-x/src/index.tsx b/node_modules/react-notion-x/src/index.tsx
index 4f651e3..5e06926 100644
--- a/node_modules/react-notion-x/src/index.tsx
+++ b/node_modules/react-notion-x/src/index.tsx
@@ -7,7 +7,6 @@ export * from './context'
 export * from './components/code'
 export * from './components/collection'
 export * from './components/collection-row'
-export * from './third-party'
 
 export { NotionRenderer } from './renderer'
 export type { NotionRendererProps } from './renderer'
diff --git a/node_modules/react-notion-x/src/renderer.tsx b/node_modules/react-notion-x/src/renderer.tsx
index 4353c49..4592c4c 100644
--- a/node_modules/react-notion-x/src/renderer.tsx
+++ b/node_modules/react-notion-x/src/renderer.tsx
@@ -74,13 +74,13 @@ export const NotionRenderer: React.FC<NotionRendererProps> = ({
   defaultPageCoverPosition,
   ...rest
 }) => {
-  const zoom =
-    typeof window !== 'undefined' &&
-    mediumZoom({
-      container: '.notion-viewport',
-      background: 'rgba(0, 0, 0, 0.8)',
-      margin: getMediumZoomMargin()
-    })
+  // const zoom =
+  //   typeof window !== 'undefined' &&
+  //   mediumZoom({
+  //     container: '.notion-viewport',
+  //     background: 'rgba(0, 0, 0, 0.8)',
+  //     margin: getMediumZoomMargin()
+  //   })
 
   return (
     <NotionContextProvider
@@ -99,7 +99,7 @@ export const NotionRenderer: React.FC<NotionRendererProps> = ({
       defaultPageIcon={defaultPageIcon}
       defaultPageCover={defaultPageCover}
       defaultPageCoverPosition={defaultPageCoverPosition}
-      zoom={zoom}
+      zoom={false}
     >
       <NotionBlockRenderer {...rest} />
     </NotionContextProvider>
